src/Api/Orbit.cpp

```diff
@@ -6,21 +6,21 @@
 #include "Api/Orbit.h"
 
 #include "Api/EncodedEvent.h"
+#include "Api/LockFreeApiEventProducer.h"
 #include "OrbitBase/Profiling.h"
 #include "OrbitBase/ThreadUtils.h"
 
-// Temporary dummy function, proper implementation is in next PR.
-// See https://github.com/google/orbit/pull/1702 for fully working prototype.
-static void EnqueueIntermediateEvent(const orbit_api::ApiEvent& event) { (void)event; }
-
 static void EnqueueApiEvent(orbit_api::EventType type, const char* name = nullptr,
                             uint64_t data = 0, orbit_api_color color = kOrbitColorAuto) {
-  uint64_t timestamp_ns = orbit_base::CaptureTimestampNs();
+  static orbit_api::LockFreeApiEventProducer producer;
```

Comment:
This statement is associated with quite some initialization work. If I get it correctly this happens right after the .so gets loaded, so it's already the case that the first call to `EnqueueApiEvent` happens together with initialization work, but I want to confirm that this is the case and it's intended.

Reply:
since this is inside the function it is going to be initialized on the first call.

Reply:
That's right. The first api call also triggers the loading of liborbit.so. We could possibly add an initialization method to explicitly trigger the loading of the .so and the initialization of the side channel if need be. Currently we have no way to unload liborbit.so manually, so having an "init" without a "deinit" is a bit weird, but it's something I'd like to add.